packages/ui/src/hooks/use-api.ts

```diff
@@ -1,98 +1,29 @@
-import type { LogtoErrorCode } from '@logto/phrases';
-import type { RequestErrorBody } from '@logto/schemas';
-import { HTTPError } from 'ky';
-import { useState, useCallback, useContext, useEffect } from 'react';
-import { useTranslation } from 'react-i18next';
+import type { Nullable } from '@silverhand/essentials';
+import { useCallback, useContext } from 'react';
 
 import { PageContext } from '@/hooks/use-page-context';
 
-type UseApi<T extends unknown[], U> = {
-  result?: U;
-  error: RequestErrorBody | undefined;
-  run: (...args: T) => Promise<U | undefined>;
-};
-
-export type ErrorHandlers = {
-  [key in LogtoErrorCode]?: (error: RequestErrorBody) => void | Promise<void>;
-} & {
-  global?: (error: RequestErrorBody) => void | Promise<void>; // Overwrite default global error handle logic
-  callback?: (error: RequestErrorBody) => void; // Callback method
-};
-
-function useApi<Args extends unknown[], Response>(
-  api: (...args: Args) => Promise<Response>,
-  errorHandlers?: ErrorHandlers
-): UseApi<Args, Response> {
-  const { t } = useTranslation();
-  const [error, setError] = useState<RequestErrorBody>();
-  const [result, setResult] = useState<Response>();
-
-  const { setLoading, setToast } = useContext(PageContext);
+const useApi = <Args extends unknown[], Response>(api: (...args: Args) => Promise<Response>) => {
```

Comment:
Then the only purpose of this hook is to manage loading state?

Reply:
After the clean up, I believe so, and safely return s a error and result.